feat(highlights): add fzf-lua

strash opened the pull request:
[https://github.com/ibhagwan/fzf-lua](https://github.com/ibhagwan/fzf-lua)

Before
<img width="1637" alt="Screenshot 2024-11-26 at 04 10 08" src="https://github.com/user-attachments/assets/8a867215-aadc-4f4e-a995-a6ed36b5787e">

After
<img width="1637" alt="Screenshot 2024-11-26 at 04 11 44" src="https://github.com/user-attachments/assets/f63db86a-2058-44fe-902d-1663b5f8a3c2">

Before
<img width="1637" alt="Screenshot 2024-11-26 at 04 11 00" src="https://github.com/user-attachments/assets/ca5c11e5-2537-47f3-945b-1d4337dbd1d4">

After
<img width="1637" alt="Screenshot 2024-11-26 at 04 12 34" src="https://github.com/user-attachments/assets/a1134a51-cc77-48ae-ba93-6c7329cfade5">

strash commented:
We should probably mention in the README about this setting for generating highlights for fzf.
```lua
require("fzf-lua").setup({ fzf_colors = true })
```

I've been using this plugin for a while now, but I just learned about this setting recently. 😀

strash commented:
Some updates.

Before 
<img width="1643" alt="Screenshot 2024-11-26 at 14 28 43" src="https://github.com/user-attachments/assets/5a87a67e-61a3-4a55-8efd-c36d843898ec">

After
<img width="1643" alt="Screenshot 2024-11-26 at 14 27 36" src="https://github.com/user-attachments/assets/4a22b7fd-219e-43cc-b8d5-abdd156c008d">

Before
<img width="1643" alt="Screenshot 2024-11-26 at 14 29 25" src="https://github.com/user-attachments/assets/6ce85231-f657-4554-8410-47d12c87f49d">

After
<img width="1643" alt="Screenshot 2024-11-26 at 14 28 07" src="https://github.com/user-attachments/assets/edaad682-aee1-4ca1-afdc-77d994da4ff5">

strash commented:
Hey @ramojus, this PR has all the changes from the PR with blink.

ramojus commented:
Hey, sorry for the delays, I'm quite busy lately.

I'm just wondering if we really need to support this plugin? We have telescope, which is much more popular. Does this plugin have anything that telescope doesn't?
Otherwise the PR looks good, I would have a few minor comments and it can be merged.

ramojus commented:
Also just saw that this plugin has a [`telescope` profile](https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#profiles). In terms of highlight groups, with your changes, does it look similar to telescope when that profile is applied?

strash commented:
> I'm just wondering if we really need to support this plugin?

[Why Fzf-Lua](https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#why-fzf-lua)
Personally, I think it's a matter of personal preference, but I find it slightly faster. I'm an Fzf user, and I would greatly appreciate it if this plugin were supported by this beautiful colorscheme. There's no real reason not to support it. By doing so, we can potentially attract more users.

---

> Also just saw that this plugin has a [telescope profile](https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#profiles). In terms of highlight groups, with your changes, does it look similar to telescope when that profile is applied?

It looks like this.

<img width="1643" alt="Screenshot 2024-12-07 at 02 06 19" src="https://github.com/user-attachments/assets/6bd37d7b-2dba-401e-a734-5bfb675063f5">
<img width="1643" alt="Screenshot 2024-12-07 at 02 07 56" src="https://github.com/user-attachments/assets/28873816-e938-4ef9-a441-91824712396e">

strash commented:
@ramojus I've removed variables

ramojus commented:
Thanks for the screenshots, that looks surprisingly good! We should make telescope look more like this :D
But of course, it's not viable unless telescope supports darker backdrop.

This is a difficult plugin as there are a lot of scenarios. I've pulled your changes to take a deeper look. There are some things I don't like about this plugin that I haven't found a way to fix:
- Current result being bold (seems unnecessary - distracting).
![image](https://github.com/user-attachments/assets/0a7c8770-269a-4bf1-b21a-fcae94dd2466)
- Backdrop removed when searching colorschemes. I get it, it's for colorscheme preview, but ideally fzf background would then be lighter when searching colorschemes.
![image](https://github.com/user-attachments/assets/3799123f-6528-4bcf-af7f-14db64c8316c)
- Line numbers in preview having their default dark background. (This is probably fine as long as the preview has `Normal` background)
![image](https://github.com/user-attachments/assets/2093356b-5e8c-4060-b06b-a203eb8bb9da)
I was thinking of making the backgrounds lighter, but with this issue, probably better to leave it as is.

As you know this plugin better than I do, I would appreciate any help with these issues. Otherwise, not a deal breaker.

I will also submit some review comments.

ramojus commented:
Nice work, really. I like that we also retain some similarity with `fzf` cli, mainly by keeping that `FzfLuaFzfPointer` visible, but also not distracting.
One more issue here:
![image](https://github.com/user-attachments/assets/2aea2ca7-0a4a-4410-ab29-602dfeb593c2)
The background of that title doesn't match with normal background. Not sure which highlight group sets it, as I don't get this title by default.

strash commented:
> * Current result being bold (seems unnecessary - distracting).
>   ![image](https://private-user-images.githubusercontent.com/41536253/395780568-0a7c8770-269a-4bf1-b21a-fcae94dd2466.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyMDIzMDksIm5iZiI6MTczNDIwMjAwOSwicGF0aCI6Ii80MTUzNjI1My8zOTU3ODA1NjgtMGE3Yzg3NzAtMjY5YS00YmYxLWIyMWEtZmNhZTk0ZGQyNDY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE0VDE4NDY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1MmE5ZGVjMGYwYTEyOWY5YzQ2MGI1MjEwODdjYThjZWMxNmJkYjMzY2Y1OTFkMTE1NmFkOTg0NGUxZTA5OTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VkaQZGxDFRZOFmMwNfHBEqf_L9bqpR-TDRkTPeZuGlw)

Not sure how to fix this. I've explicitly unset bold and set nocombine, but it seems to be ignored.  

> * Backdrop removed when searching colorschemes. I get it, it's for colorscheme preview, but ideally fzf background would then be lighter when searching colorschemes.
>   ![image](https://private-user-images.githubusercontent.com/41536253/395780595-3799123f-6528-4bcf-af7f-14db64c8316c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyMDIzMDksIm5iZiI6MTczNDIwMjAwOSwicGF0aCI6Ii80MTUzNjI1My8zOTU3ODA1OTUtMzc5OTEyM2YtNjUyOC00YmNmLWFmN2YtMTRkYjY0YzgzMTZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE0VDE4NDY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAzZjQxMjA3NjY0ODUzY2YzODQxMGI0NDAyMmE2MjdlOWUzNjEzYTE4YTAxZDIzMzMxNTM4MTU1MjliOGI5YmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5fc-cTMpIFxuv9FjpD-JwEjpu8Plp0lvwg8hB9lriMU)

I think it's fine. And probably there's nothing we can do about it.

> * Line numbers in preview having their default dark background. (This is probably fine as long as the preview has `Normal` background)
>   ![image](https://private-user-images.githubusercontent.com/41536253/395780612-2093356b-5e8c-4060-b06b-a203eb8bb9da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyMDIzMDksIm5iZiI6MTczNDIwMjAwOSwicGF0aCI6Ii80MTUzNjI1My8zOTU3ODA2MTItMjA5MzM1NmItNWU4Yy00MDYwLWIwNmItYTIwM2ViOGJiOWRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE0VDE4NDY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3MDc4MTdmYWIxYTBjNTYzZDA2NzEzM2VhOTFiNmYyNGQyNmYwZGEyYTRkY2MwNGVjOWIxMDRiMjY3NjU2MGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tcZc3JR2H_IHPXJveksERTvLs4SbF9pOcXnnDkSv5ic)
>   I was thinking of making the backgrounds lighter, but with this issue, probably better to leave it as is.

Fixed.

> One more issue here: 
> ![image](https://private-user-images.githubusercontent.com/41536253/395781322-2aea2ca7-0a4a-4410-ab29-602dfeb593c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyMDIzMDksIm5iZiI6MTczNDIwMjAwOSwicGF0aCI6Ii80MTUzNjI1My8zOTU3ODEzMjItMmFlYTJjYTctMGE0YS00NDEwLWFiMjktNjAyZGZlYjU5M2MyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE0VDE4NDY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4MDQxM2EwYzI4NzIwMDlmY2IyZWM4YzI2OGM1YmY4NmZhNjg3MDM5NzM1Mzg1NWNlMzg3MmQ0Nzc2OTI3YzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fuArsFghtEMyyAhqFMU48lh7BjE9qIZMlAeviaykISA)
> The background of that title doesn't match with normal background. Not sure which highlight group sets it, as I don't get this title by default.

Nice catch! Fixed.

strash commented on lua/mellifluous/highlights/plugins/fzf.lua:
```diff
@@ -0,0 +1,36 @@
+    hl.set("FzfLuaPreviewNormal", { bg = hl.get("Normal").bg, fg = hl.get("Normal").fg })
+    hl.set("FzfLuaPreviewBorder", { bg = hl.get("Normal").bg, fg = hl.get("Normal").bg })
+    hl.set("FzfLuaPreviewTitle", { link = "FzfLuaTitle" })
+    hl.set("FzfLuaCursorLine", { bg = colors.bg2, fg = colors.fg2 })
```
Oh, apparently, `FzfLuaCursorLine` is intended for the preview window. I was confused about the naming convention and removed it.

ramojus replied:
Oh shit, you're right, it's for preview, I also thought it's for search selection. But `FzfLuaFzfCursorLine` links to this group by default. This suggestion then applies to `FzfLuaFzfCursorLine` group:
```lua
    hl.set("FzfLuaFzfCursorLine", { bg = groups.MenuButtonSelected(colors.dark_bg).bg, fg = colors.fg3 })
```

strash replied:
<img width="719" alt="image" src="https://github.com/user-attachments/assets/086b861e-fe0c-4a43-9dab-1d364d66b55a" />

strash commented:
<img width="1643" alt="Screenshot 2024-12-14 at 23 51 09" src="https://github.com/user-attachments/assets/fd223ccf-b4a4-49db-9f06-957829527eb3" />
<img width="1643" alt="Screenshot 2024-12-14 at 23 51 51" src="https://github.com/user-attachments/assets/ac6779e6-af15-4ac9-9539-44cef9ad984a" />

ramojus commented:
Thanks!
